token-client/src/main/java/com/sap/cloud/security/client/HttpClientFactory.java

```diff
@@ -48,6 +50,16 @@ static CloseableHttpClient create(ClientIdentity clientIdentity) throws HttpClie
 		if (services.isEmpty()) {
 			throw new ProviderNotFoundException("No HttpClientFactory service could be found in the classpath");
 		}
+		if (services.size() > 2) {
+			throw new ProviderException(
+					"More than 2 HttpClientFactory service providers found. There should be only one");
+		}
+		if (services.size() == 2) {
+			return services.stream()
+					.filter(httpClientFactory -> !httpClientFactory.getClass().getName()
+							.equals("com.sap.cloud.security.client.DefaultHttpClientFactory"))
```

Comment:
would be cool, if we have a constant for that class name as part of the DefaultHttpClientFactory class. Better for refactoring.